data/catalogs/test_data_catalog.py

```diff
@@ -0,0 +1,117 @@
+"""A command line script for testing predefined data catalogs.
+
+This script allows for testing separate data catalog items and the complete data catalog.
+When testing the data catalog all the paths of the data catalog sources are
+checked if they exist.
+Call for testing the data catalog:
+    'python test_data_catalog.py path/to/datacatalog.yml'
+
+Separate data catalog sources can be tested by giving the dataset source name as
+an extra argument. The argument should be preceded by -ds. It is also possible to specify
+the version of the dataset source by supplying the call with a -dsv flag and version
+number. The dataset is opened and if the dataset contains geo data the dataset is clipped
+to a small bounding box surrounding Deltares Delft office.
+Example dataset source test call:
+    'python test_data_catalog.py path/to/datacatalog.yml -ds chelsa -dsv 1.2
+    -r "{'bbox':[4.333409,51.962159,4.42835,52.006873]}"'
+
+In addition the passed data catalog yaml is checked if it is a valid data catalog yaml.
+
+"""
+import argparse
+import json
+
+from dask.distributed import Client
+
+from hydromt import DataCatalog
+from hydromt.log import setuplog
+from hydromt.validators.data_catalog import DataCatalogValidator
+
+client = Client(processes=False)
+logger = setuplog()
+
+
+def test_dataset(args, datacatalog):
+    """Tests a given dataset on opening the dataset and minimal processing.
+
+    Datasets containing geo data will be clipped to a small bounding box. This bounding box can
+    be given by a user with the -bbox flag.
+    """
+    source = datacatalog.get_source(source=args.dataset, version=args.dataset_version)
+    if args.region:
+        region_json = args.region.replace("'", '"')
+        bbox = json.loads(region_json)["bbox"]
+        print(bbox)
+    else:
+        bbox = [
+            4.333409,
+            51.962159,
+            4.42835,
+            52.006873,
+        ]  # small area surrounding Deltares Delft office
+    logger.info(
+        f"Retrieving {args.dataset} {source.data_type} and clipping by {bbox} bounding box"
+    )
+    if source.data_type == "GeoDataFrame":
+        dataset = datacatalog.get_geodataframe(args.dataset, bbox=bbox)
+    elif source.data_type == "GeoDataset":
+        dataset = datacatalog.get_geodataset(args.dataset, bbox=bbox)
+    elif source.data_type == "RasterDataset":
+        dataset = datacatalog.get_rasterdataset(args.dataset, bbox=bbox)
+    elif source.data_type == "DataFrame":
+        dataset = datacatalog.get_dataframe(args.dataset)
+    elif source.data_type == "Dataset":
+        dataset = datacatalog.get_dataset(args.dataset)
+    assert dataset is not None
+
+
+def test_data_catalog(args, datacatalog):
+    """Tests the paths of the given data catalog."""
+    error_count = 0
+    logger.info("Checking paths of data catalog sources")
+    for source in datacatalog.get_source_names():
```

Comment:
I think the source manes might not be loaded at this point yet (because we lazy load sources iirc) if you checked it  and this returns all the source names, that's fine and you can just close this comment, if not please double check, it would be a shame if the script didn't actually do anything.

Reply:
get_source_names works fine :)